src/transformers/integrations/flash_paged.py

```diff
@@ -50,24 +49,35 @@ def paged_attention_forward(
         window_size: (left, right). If not (-1, -1), implements sliding window local attention.
         softcap: float. Anything > 0 activates softcapping attention.
     """
-    k, v = cache.update(k, v, module.layer_idx, **kwargs)
+    # .update changes the shape of k and v from [1, num_kv_heads, seqlen_kv, head_dim] to [-1, num_kv_heads, head_dim]
+    if cache is not None:
+        k, v = cache.update(k, v, module.layer_idx, **kwargs)
+
+        # Check if we are in a sliding window context
+        is_full_attention = cache.sliding_windows[module.layer_idx] == NO_SLIDING_WINDOW
+        cu_seq_lens_k = cu_seq_lens_k[0].clone() if is_full_attention else cu_seq_lens_k[1].clone()
+        max_seqlen_k = max_seqlen_k[0] if is_full_attention else max_seqlen_k[1]
```

Comment:
same, for explicitly we would have a dict maybe no? instead of a 2 index list?

Reply:
layer_type no?

Reply:
Something like that!